### PR TITLE
feat: CouchDbStream Implementation

### PR DESCRIPTION
**Pull request by rjtmahinay**
Implementation of streams for CouchDb using Vert.x ReadStream

### CODE REVIEW

**Comment by rjtmahinay** [comment]
@Stwissel @jmgang Showing to you the initial implementation of CouchDbStream

File: src/main/java/io/vertx/ext/couchdb/streams/CouchDbStream.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+package io.vertx.ext.couchdb.streams;
+
+import io.vertx.codegen.annotations.VertxGen;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.streams.ReadStream;
+import io.vertx.core.streams.WriteStream;
+
+@VertxGen
```
Comment (Stwissel) [inline src/main/java/io/vertx/ext/couchdb/streams/CouchDbStream.java:20]: Some JavaDoc would be helpful

File: src/main/java/io/vertx/ext/couchdb/streams/impl/CouchDbStreamImpl.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    return this;
+  }
+
+  @Override
+  public CouchDbStream endHandler(Handler<Void> endHandler) {
+    this.endHandler = endHandler;
+    return this;
+  }
+
+  private void doFetch() {
```
Comment (Stwissel) [inline src/main/java/io/vertx/ext/couchdb/streams/impl/CouchDbStreamImpl.java:82]: ??

**Review by Stwissel — APPROVED** [review]
What's the use case for this stream?

**Comment by Stwissel** [comment]
No quite sure where that fits. To handle incoming streams (e.g. change stream or chunked encoding we need a writeStream. E..g.

```java
  client
        .get(5984, "localhost", path)
        .authentication(getCredentials())
        .as(BodyCodec.pipe(writeStream))
        .send()
        .expecting(HttpResponseExpectation.SC_SUCCESS)
        .onSuccess(....)
``` 

where writestream gets the data

File: src/main/java/io/vertx/ext/couchdb/streams/CouchDbStream.java (hunk 1 above)
Comment (rjtmahinay) [reply src/main/java/io/vertx/ext/couchdb/streams/CouchDbStream.java:20]: I'll add javadoc

**Review by jmgang — APPROVED** [review]

**Pull request merged**